cas-server-webapp-support/src/main/java/org/jasig/cas/util/CasLoggerContextInitializer.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Licensed to Apereo under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work
+ * for additional information regarding copyright ownership.
+ * Apereo licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License.  You may obtain a
+ * copy of the License at the following location:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.jasig.cas.util;
+
+import org.apache.commons.lang3.StringUtils;
+import org.reflections.Reflections;
+import org.reflections.scanners.SubTypesScanner;
+import org.reflections.util.ClasspathHelper;
+import org.reflections.util.ConfigurationBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.io.Resource;
+import org.springframework.stereotype.Component;
+import org.springframework.web.context.ServletContextAware;
+
+import javax.annotation.PreDestroy;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import javax.validation.constraints.NotNull;
+import java.net.URL;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Initializes the CAS logging framework by calling
+ * the logger initializer and sets the location of the
+ * log configuration file.
+ * @author Misagh Moayyed
+ * @since 4.1
+ */
+@Component
+public final class CasLoggerContextInitializer implements ServletContextAware {
```

Comment:
Why do we need this class?

I think adding custom (logging) configuration classes is a bad idea. Because of the last CasLoggerFactory class to change TGT values on the fly for logging, I haven't been able to start the CAS server through the Maven Jetty plugin.